app/controllers/api/v1/appointment_slots_controller.rb

```diff
@@ -0,0 +1,87 @@
+module Api
+  module V1
+    class AppointmentSlotsController < Api::V1::ApiController
+      load_and_authorize_resource :appointment_slot, parent: false
+
+      resource_description do
+        short 'Manage a list of appointment quotas for special days'
+        formats ['json']
+        error 401, "Unauthorized"
+        error 404, "Not Found"
+        error 422, "Validation Error"
+        error 500, "Internal Server Error"
+      end
+
+      def_param_group :appointment_slot do
+        param :appointment_slot, Hash, required: true do
+          param :timestamp, String
+          param :note, String
+          param :quota, :number
+        end
+      end
+
+      def appointment_slot_params
+        params.require(:appointment_slot).permit([:timestamp, :quota, :note])
+      end
+
+      api :GET, '/v1/appointment_slots', "List upcoming appointment slots"
+      def index
+        render_with_timezone(@appointment_slots.upcoming.ascending)
+      end
+
+      api :GET, '/v1/appointment_slots/calendar', "List upcoming appointment slots aggregated by dates"
+      def calendar
+        from = params[:from] ? Date.parse(params[:from]) : Date.today
+        to = Date.parse(params[:to])
```

Comment:
Might be good to put an upper limit on the `to` field as this is a user entered param. Can avoid denial of service type attacks where someone enters to 2222-12-31 and we end up iterating every day between now and then in ```AppoinmentSlot.calendar```

Reply:
good point !